pkgs/development/compilers/terra/default.nix

```diff
@@ -1,61 +1,70 @@
 { lib, stdenv, fetchurl, fetchFromGitHub
-, llvmPackages, ncurses, lua
+, llvmPackages, ncurses, cmake, libxml2, symlinkJoin
 }:
 
 let
-  luajitArchive = "LuaJIT-2.0.5.tar.gz";
+  luajitRev = "9143e86498436892cb4316550be4d45b68a61224";
+  luajitArchive = "LuaJIT-${luajitRev}.tar.gz";
   luajitSrc = fetchurl {
-    url    = "http://luajit.org/download/${luajitArchive}";
-    sha256 = "0yg9q4q6v028bgh85317ykc9whgxgysp76qzaqgq55y6jy11yjw7";
+    url = "https://github.com/LuaJIT/LuaJIT/archive/${luajitRev}.tar.gz";
+    sha256 = "0kasmyk40ic4b9dwd4wixm0qk10l88ardrfimwmq36yc5dhnizmy";
   };
 in
 stdenv.mkDerivation rec {
   pname = "terra";
-  version = "1.0.0pre1175_${builtins.substring 0 7 src.rev}";
+  version = "1.0.0-beta3";
 
   src = fetchFromGitHub {
-    owner  = "zdevito";
+    owner  = "terralang";
     repo   = "terra";
-    rev    = "ef6a75ffee15a30f3c74f4e6943851cfbc0fec3d";
-    sha256 = "0aky17vbv3d9zng34hp17p9zb00dbzwhvzsdjzrrqvk9lmyvix0s";
+    rev    = "release-${version}";
+    sha256 = "15ik32xnwyf3g57jvvaz24f6a18lv3a86341rzjbs30kd5045qzd";
   };
 
-  nativeBuildInputs = [ lua ];
-  buildInputs = with llvmPackages; [ llvm clang-unwrapped ncurses ];
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    (symlinkJoin {
+      name = "llvmClangMerged";
+      paths = with llvmPackages; [ llvm clang-unwrapped ];
+    })
+    ncurses
+    libxml2
+  ];
+
+  cmakeFlags = [
+    "-DHAS_TERRA_VERSION=0"
+    "-DTERRA_VERSION=${src.rev}"
+    "-DTERRA_LUA=luajit"
+  ];
 
   doCheck = true;
   enableParallelBuilding = true;
   hardeningDisable = [ "fortify" ];
   outputs = [ "bin" "dev" "out" "static" ];
 
-  patches = [ ./nix-cflags.patch ];
+  patches = [
+    # Should be removed as https://github.com/terralang/terra/pull/496 get merged and released.
+    ./get-compiler-from-envvar-fix-cpu-detection.patch
+    ./nix-cflags.patch
+    ./disable-luajit-file-download.patch
+  ];
   postPatch = ''
-    substituteInPlace Makefile \
-      --replace '-lcurses' '-lncurses'
-
     substituteInPlace src/terralib.lua \
       --subst-var-by NIX_LIBC_INCLUDE ${lib.getDev stdenv.cc.libc}/include
```

Comment:
This is also going to need to be tweaked in light of the LLVM output split. I'm still identifying what needs to be done with this one.

Reply:
What specifically goes wrong here is that I start seeing missing `stddef.h` errors in the tests, which is an INCLUDE_PATH problem.

Reply:
Any news? I'm stuck with this too.

Reply:
Okay, figured out how I'm doing this. The issue is that CMake does not pass outside environment variables into tests unless they're specified in the test properties. I'm putting together a substitution to add it in. Should have it for you later today.

Reply:
Okay, getting closer but don't have it yet. Current roadblock is figuring out a derivation merge that has `stddef.h` in an appropriate place, but that should be straightforward enough.

Reply:
Hah, nailed it. Okay, do you want me to write it in a comment here, put it in a PR, or add it as a review? Also, given that I am actually paid to make Terra stop being stupid on the regular, do we want to add me to the maintainers list for this thing?

Reply:
I think it's better to create a new PR.

Reply:
Abandoning this one and making a new one that I'll submit, or PRing against yours to pull into this one?

Reply:
I'm ok with the first option. I'll close this PR and you open a new one.

Reply:
Okay. Shiny. I'll set that up shortly. You want to be included in the maintainer list?

Reply:
No. I've learned about terra because ZHF so I have almost zero knowledge of it. Thanks for asking.